src/app/(auth)/sign-in/components/NewPasswordContainer.tsx

```diff
@@ -29,14 +36,23 @@ function NewPasswordContainer({ onClick }: NewPasswordContainerProps) {
     resolver: zodResolver(validationSchema),
   });
 
-  const onSubmit: SubmitHandler<ValidationSchema> = (data) => {
-    console.log(data);
+  const onSubmit: SubmitHandler<ValidationSchema> = async (data) => {
     onClick();
+    try {
+      if (token) {
+        // TODO: will need to refactor once we can figure out the issue with containerState
+        await resetPassword(data.password, token).then(() =>
+          router.push("/sign-in"),
+        );
+      }
+    } catch (error) {
+      console.error("Error to change password: ", error);
+    }
```

Comment:
None of this try catch stuff is necessary, please follow the pattern used in the project related to handling async requests. For the error case, we should be displaying the modal with the error, not a console.error.